### PR TITLE
Overloading the set_thread_affinity method for Windows compatibility

### PR DESCRIPTION
**Pull request by SENAI-GilmarCorreia**
Just me and my Windows setup again :)

### CODE REVIEW

File: include/realtime_tools/realtime_helpers.hpp [hunk 1]
```diff
@@ -29,6 +29,7 @@
 #ifndef REALTIME_TOOLS__REALTIME_HELPERS_HPP_
 #define REALTIME_TOOLS__REALTIME_HELPERS_HPP_
 
+#include <pthread.h>
```
Comment (saikishor) [inline include/realtime_tools/realtime_helpers.hpp:32]: @SENAI-GilmarCorreia Do you need `pthread.h` here?

Comment (SENAI-GilmarCorreia) [reply include/realtime_tools/realtime_helpers.hpp:32]: To compile successfully, I really need this.

If it’s not included, I get the following error:

```
C:\path\to\workspace\src\realtime_tools\include\realtime_tools/realtime_helpers.hpp(72,50): error C2065: 'pthread_t': undeclared identifier [C:\path\to\workspace\build\realtime_tools\realtime_tools.vcxproj]
C:\path\to\workspace\src\realtime_tools\include\realtime_tools/realtime_helpers.hpp(72,60): error C2146: syntax error: missing ')' before identifier 'thread' [C:\path\to\workspace\build\realtime_tools\realtime_tools.vcxproj]
  Building Custom Rule C:/path/to/workspace/src/realtime_tools/CMakeLists.txt
  realtime_helpers.cpp
C:\path\to\workspace\src\realtime_tools\include\realtime_tools/realtime_helpers.hpp(72,50): error C2065: 'pthread_t': undeclared identifier [C:\path\to\workspace\build\realtime_tools\thread_priority.vcxproj]
C:\path\to\workspace\src\realtime_tools\include\realtime_tools/realtime_helpers.hpp(72,60): error C2146: syntax error: missing ')' before identifier 'thread' [C:\path\to\workspace\build\realtime_tools\thread_priority.vcxproj]
```

Comment (saikishor) [reply include/realtime_tools/realtime_helpers.hpp:32]: So, you had to add it for pthread_t. I've a good solution to this then. I'll try to add it to your PR when I get some time 

So, we can avoid that header for windows

File: src/realtime_helpers.cpp [hunk 2]
```diff
@@ -115,13 +115,16 @@ bool lock_memory(std::string & message)
 #endif
 }
 
-std::pair<bool, std::string> set_thread_affinity(pthread_t thread, int core)
-{
-  std::string message;
 #ifdef _WIN32
-  message = "Thread affinity is not supported on Windows.";
+std::pair<bool, std::string> set_thread_affinity(HANDLE thread, int core)
```
Comment (saikishor) [inline src/realtime_helpers.cpp:119]: @SENAI-GilmarCorreia Don't you need a definition of this method in the header file?

**Comment by saikishor** [comment]
@SENAI-GilmarCorreia Do you mind if I push some changes to your branch?

**Comment by SENAI-GilmarCorreia** [comment]
No problem! @saikishor

File: src/realtime_helpers.cpp (hunk 2 above)
Comment (SENAI-GilmarCorreia) [reply src/realtime_helpers.cpp:119]: let me see If I can change this.

Comment (saikishor) [reply src/realtime_helpers.cpp:119]: No need. I'm already working on it now. I'll push it in few min

**Comment by SENAI-GilmarCorreia** [comment]
@saikishor Done!

**Comment by SENAI-GilmarCorreia** [comment]
@saikishor I've just fixed the issue by using the `__unix__` flag instead of `UNIX`, which wasn't defined correctly for non-Windows environments.

File: src/realtime_helpers.cpp [hunk 3]
```diff
@@ -28,9 +28,7 @@
 
 #include "realtime_tools/realtime_helpers.hpp"
 
-#ifdef _WIN32
-#include <windows.h>
-#else
+#ifdef __unix__
```
Comment (aprotyas) [inline src/realtime_helpers.cpp:31]: I don't know which platforms we end up targeting or want to support, but [just leaving some text here](https://web.archive.org/web/20140625123925/http://nadeausoftware.com/articles/2012/01/c_c_tip_how_use_compiler_predefined_macros_detect_operating_system#:~:text=However%2C%20compilers%20for%20OSX%2C%20iOS%2C%20and%20Darwin%20%20do%20not%20define%20__unix__.%20To%20detect%20all%20BSD%20OSes%2C%20including%20OSX%2C%20iOS%2C%20and%20Darwin%2C%20use%20an%20#if/#endif%20that%20checks%20for%20__unix__%20along%20with%20__APPLE__%20and%20__MACH__%20(see%20the%20later%20section%20on%20OSX%20and%20iOS).) to ruminate on:

> However, compilers for OSX, iOS, and Darwin do not define `__unix__`. To detect all BSD OSes, including OSX, iOS, and Darwin, use an `#if`/`#endif` that checks for `__unix__` along with `__APPLE__` and `__MACH__` (see the later section on OSX and iOS).

Comment (saikishor) [reply src/realtime_helpers.cpp:31]:
```suggestion
#if defined(__unix__) || (defined(__APPLE__) && defined(__MACH__))
```
Make sense. I think we can use the following one as in the link you suggested  @aprotyas

**Comment by saikishor** [comment]
@SENAI-GilmarCorreia I see we mention only `_WIN32`, should we also handle `_WIN64`?

**Comment by SENAI-GilmarCorreia** [comment]
> @SENAI-GilmarCorreia I see we mention only `_WIN32`, should we also handle `_WIN64`?

Hello @saikishor. I have seen that _WIN32 supports both 32-bit and 64-bit Windows platforms in Microsoft Visual Studio.

**Review by saikishor — APPROVED** [review]

**Review by christophfroehlich — APPROVED** [review]

**Pull request merged**